### ISSUE TITLE
Screen Recording/ Can't capture long videos

### ISSUE THREAD

**Issue by makingbillions** [ShareX/ShareX#1743]
Hi, 

Love this little tool. It's a breeze to use and is so functional. Thank you to all the developers :-)

I am trying to screen record a long webinar - 2hrs plus. 

What's happening at the moment is that it crashes and I am unable to recover the screen recording and It wont let me stop the video.  

Thanks

**Comment by mjamesb80** [comment]
I have the same issue. I am using the portable version. Have you had any luck fixing this issue?

**Comment by makingbillions** [comment]
No. I am using camtasiac recorder instead (paid).

**Comment by Jaex** [comment]
FFmpeg doing recording so I don't think I can fix it. Also mainly when error happen in FFmpeg, ShareX shows output log and without be able to see that log I can't know what error it gave too.

**Comment by mjamesb80** [comment]
I added a copy of the output log along with the error box afterwards. 
[Error file.txt](https://github.com/ShareX/ShareX/files/389124/Error.file.txt)

So the FFmpeg is doing the recording and that is the problem? 

![capture](https://cloud.githubusercontent.com/assets/20707377/17225345/2c2d0afe-54ca-11e6-9939-4e143e931694.JPG)

**Comment by Scrxtchy** [comment]
You might need to optimise your ffmpeg arguments for really long recordings.
One of my university lectures does all his recordings with ffmpeg on linux, so I can say for sure it's possible.

However, as mentioned, this isn't a direct issue with sharex itself, all that is suggested is to use a custom command. 
![](http://jesusfuck.me/di/FG2O/16-07-29-15-46-32-screen-recording-cant-capture-long-videos-issu.png)
If I can track down an effective solution to ffmpeg, I will post it here for future reference

**Comment by Jaex** [comment]
Now I have same issue too, unable to stop it and when stop button clicked 3 times then it kills ffmpeg process and there is no error in output too.

**Comment by Jaex** [comment]
Were you using Windows 10 anniversary update?

**Comment by mjamesb80** [comment]
I am using Windows 10 anniversary build
Today, I changed the video codec from x264(mp4) to x265(mp4) and was successful in recording a 45 minute session. I'll include a screen shot of my settings used. Let me know if you are able to record successfully with similar settings. 

![screen recording options](https://cloud.githubusercontent.com/assets/20707377/17405793/8efca8c2-5a26-11e6-8ebf-e32c9d386d3a.jpg)

**Comment by Collisionc** [comment]
Try this version to see if it fixes your issue: http://collision.heroicstrike.org/ffmpeg.exe
(It is gpl-compliant)

I used different options and a older GCC version when building that, and it seems to work on my computer while the version that is downloaded via sharex crashes like you all are mentioning.

Script used for building: https://github.com/rdp/ffmpeg-windows-build-helpers
(cross-compiled linux --> windows)

ffmpeg version N-81426-geb5f4b1 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 4.9.3 (GCC)
  configuration: --arch=x86_64 --target-os=mingw32 --pkg-config=pkg-config --disable-w32threads --enable-libsoxr --enable-fontconfig --enable-libass --enable-libbluray --enable-iconv --enable-libtwolame --extra-cflags=-DLIBTWOLAME_STATIC --enable-libzvbi --enable-libcaca --enable-libmodplug --extra-libs=-lstdc++ --extra-libs=-lpng --enable-decklink --extra-libs=-loleaut32 --enable-libmp3lame --enable-version3 --enable-zlib --enable-librtmp --enable-libvorbis --enable-libtheora --enable-libspeex --enable-libopenjpeg --enable-gnutls --enable-libgsm --enable-libfreetype --enable-libopus --enable-bzlib --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libvo-amrwbenc --enable-libschroedinger --enable-libvpx --enable-libilbc --enable-libwavpack --enable-libwebp --enable-libgme --enable-dxva2 --enable-avisynth --enable-gray --enable-libopenh264 --enable-nvenc --enable-libebur128 --enable-netcdf --enable-libflite --enable-lzma --enable-libsnappy --enable-libzimg --enable-gpl --enable-libx264 --enable-libx265 --enable-frei0r --enable-filter=frei0r --enable-librubberband --enable-libvidstab --enable-libxavs --enable-libxvid --enable-libmfx --extra-libs=-lpsapi --extra-libs=-lspeexdsp --extra-cflags='-mtune=core2' --extra-cflags=-O3 --enable-static --disable-shared  --enable-runtime-cpudetect
  libavutil      55. 29.100 / 55. 29.100
  libavcodec     57. 54.100 / 57. 54.100
  libavformat    57. 47.101 / 57. 47.101
  libavdevice    57.  0.102 / 57.  0.102
  libavfilter     6. 52.100 /  6. 52.100
  libswscale      4.  1.100 /  4.  1.100
  libswresample   2.  1.100 /  2.  1.100
  libpostproc    54.  0.100 / 54.  0.100


Version from zeanoe (Doesn't work)

ffmpeg version N-81423-g61fac0e Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 5.4.0 (GCC)
  configuration: --enable-gpl --enable-version3 --disable-w32threads --enable-dxva2 --enable-libmfx --enable-nvenc --enable-avisynth --enable-bzlib --enable-libebur128 --enable-fontconfig --enable-frei0r --enable-gnutls --enable-iconv --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libfreetype --enable-libgme --enable-libgsm --enable-libilbc --enable-libmodplug --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopus --enable-librtmp --enable-libschroedinger --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvo-amrwbenc --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzimg --enable-lzma --enable-decklink --enable-zlib
  libavutil      55. 29.100 / 55. 29.100
  libavcodec     57. 54.100 / 57. 54.100
  libavformat    57. 47.101 / 57. 47.101
  libavdevice    57.  0.102 / 57.  0.102
  libavfilter     6. 52.100 /  6. 52.100
  libswscale      4.  1.100 /  4.  1.100
  libswresample   2.  1.100 /  2.  1.100
  libpostproc    54.  0.100 / 54.  0.100

**Comment by shareXfan** [comment]
I can confirm that replacing existing ffmpeg.exe in **tools** folder with [http://collision.heroicstrike.org/ffmpeg.exe](http://collision.heroicstrike.org/ffmpeg.exe) provided by **collisionc** fixed my problem with crashing screen recorder.

THANX A LOT!

**Comment by vectorwarrior** [comment]
Which 'tools' folder are we supposed to put it in? There isn't one in my ShareX folder and I can't find any FFMPEG folders around either. Thanks :)

**Comment by Collisionc** [comment]
all you have to do is point sharex at it, go to task settings, screen recorder, "screen recording options" and then change the ffmpeg path so that it selects the one that you downloaded.

**Comment by vectorwarrior** [comment]
Derp, should have seen that. Thanks :+1:

**Comment by dataforge** [comment]
try adding -f ismv -movflags frag_keyframe to the screen recorder additional command lines

**Comment by John3** [comment]
I can confirm that replacing the ffmpeg.exe provided by collisionc fixed the problem.

**Comment by DaveyEdwards** [comment]
I have the current version and replacing the ffmpeg.exe worked for me!

For those like me who don't like using their brain as little as possible navigating a chat:

Right click the ShareX icon in the bottom right > Go to Task Settings > Screen Recorder > Screen recording options.. > In the top path it will show you the folder where your ffmpeg.exe is currently installed, navigate to that folder and paste in the exe you downloaded from http://collision.heroicstrike.org/ffmpeg.exe
Then close ShareX and reopen it, voila!  I swear the quality went up when doing this to

**Comment by TNTUP** [comment]
Does not fix my issue (guessing im the only one who have it) when I start the GIF recorder, the countdown doesn't start. Still at 00:00, should be at 00:15 (15 seconds). Tried that ffmpeg from heroicstrike, no dice. 

Had to record my issue with Action! what I mean: https://i.qcfb.ca/Video%202017-05-17%2023-41-57.mp4.

GDI Grab seems to bug with me, but when I try the screen-capture-recorder, quite the same but I notice at the end the green square and disappears. Encodes, then upload the final GIF and it works! Just the recorder status giving no status while recording. Can't remember since when it happens :/

**Comment by psychoadept** [comment]
I too am having an error with GIF recording.  It doesn't happen every time, but seems to be most.  Started after I had to reconfigure my hotkeys recently because for some reason it wasn't doing GIF recordings anymore and I had to make it do them again.  I'm getting these error messages:

`ffmpeg version N-81950-g7ba7b96 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 4.9.3 (GCC)
  configuration: --arch=x86_64 --target-os=mingw32 --cross-prefix=/opt/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/bin/x86_64-w64-mingw32- --pkg-config=pkg-config --disable-w32threads --enable-libsoxr --enable-fontconfig --enable-libass --enable-libbluray --enable-iconv --enable-libtwolame --extra-cflags=-DLIBTWOLAME_STATIC --enable-libzvbi --enable-libcaca --enable-libmodplug --extra-libs=-lstdc++ --extra-libs=-lpng --enable-decklink --extra-libs=-loleaut32 --enable-libmp3lame --enable-version3 --enable-zlib --enable-librtmp --enable-libvorbis --enable-libtheora --enable-libspeex --enable-libopenjpeg --enable-gnutls --enable-libgsm --enable-libfreetype --enable-libopus --enable-bzlib --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libvo-amrwbenc --enable-libschroedinger --enable-libvpx --enable-libilbc --enable-libwavpack --enable-libwebp --enable-libgme --enable-dxva2 --enable-avisynth --enable-gray --enable-libopenh264 --enable-libebur128 --enable-netcdf --enable-libflite --enable-lzma --enable-libsnappy --enable-libzimg --enable-gpl --enable-libx264 --enable-libx265 --enable-frei0r --enable-filter=frei0r --enable-librubberband --enable-libvidstab --enable-libxavs --enable-libxvid --enable-libmfx --extra-libs=-lpsapi --extra-libs=-lspeexdsp --extra-cflags='-mtune=core2' --extra-cflags=-O3 --enable-static --disable-shared --prefix=/opt/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32 --enable-runtime-cpudetect
  libavutil      55. 32.100 / 55. 32.100
  libavcodec     57. 61.100 / 57. 61.100
  libavformat    57. 51.103 / 57. 51.103
  libavdevice    57.  0.102 / 57.  0.102
  libavfilter     6. 63.100 /  6. 63.100
  libswscale      4.  1.100 /  4.  1.100
  libswresample   2.  2.100 /  2.  2.100
  libpostproc    54.  0.100 / 54.  0.100
[gdigrab @ 000002e4ffcc9100] Capture area (-1659,53),(-607,883) extends outside window area (-1536,0),(1920,1080)desktop: I/O error
ffmpeg version N-81950-g7ba7b96 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 4.9.3 (GCC)
  configuration: --arch=x86_64 --target-os=mingw32 --cross-prefix=/opt/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/bin/x86_64-w64-mingw32- --pkg-config=pkg-config --disable-w32threads --enable-libsoxr --enable-fontconfig --enable-libass --enable-libbluray --enable-iconv --enable-libtwolame --extra-cflags=-DLIBTWOLAME_STATIC --enable-libzvbi --enable-libcaca --enable-libmodplug --extra-libs=-lstdc++ --extra-libs=-lpng --enable-decklink --extra-libs=-loleaut32 --enable-libmp3lame --enable-version3 --enable-zlib --enable-librtmp --enable-libvorbis --enable-libtheora --enable-libspeex --enable-libopenjpeg --enable-gnutls --enable-libgsm --enable-libfreetype --enable-libopus --enable-bzlib --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libvo-amrwbenc --enable-libschroedinger --enable-libvpx --enable-libilbc --enable-libwavpack --enable-libwebp --enable-libgme --enable-dxva2 --enable-avisynth --enable-gray --enable-libopenh264 --enable-libebur128 --enable-netcdf --enable-libflite --enable-lzma --enable-libsnappy --enable-libzimg --enable-gpl --enable-libx264 --enable-libx265 --enable-frei0r --enable-filter=frei0r --enable-librubberband --enable-libvidstab --enable-libxavs --enable-libxvid --enable-libmfx --extra-libs=-lpsapi --extra-libs=-lspeexdsp --extra-cflags='-mtune=core2' --extra-cflags=-O3 --enable-static --disable-shared --prefix=/opt/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32 --enable-runtime-cpudetect
  libavutil      55. 32.100 / 55. 32.100
  libavcodec     57. 61.100 / 57. 61.100
  libavformat    57. 51.103 / 57. 51.103
  libavdevice    57.  0.102 / 57.  0.102
  libavfilter     6. 63.100 /  6. 63.100
  libswscale      4.  1.100 /  4.  1.100
  libswresample   2.  2.100 /  2.  2.100
  libpostproc    54.  0.100 / 54.  0.100
C:\Users\scampbll\Documents\ShareX\Screenshots\2017-05\2017-05-27_16-56-56.mp4: No such file or directory
`

**Comment by madprops** [comment]
I started getting this error with the Sharex provided ffmpeg when recording with sound http://i.imgur.com/r6JKx2s.jpg

This started happening right after installing the Creators Update in Windows 10. 

I tried using the Collision ffmpeg and the error doesn't appear anymore BUT the video and sound ended being very out of sync for some reason.

This is how I have the recording options set up http://i.imgur.com/S8CKsUJ.jpg

**Comment by madprops** [comment]
Thanks to Scrxtchy's suggestion in IRC I fixed the audio and video being out of sync by "resetting" the sources, that is, putting them in None and then selecting the same ones again. Everything is working perfectly now, with the Collision ffmpeg. I tried the Sharex ffmpeg again and got the same old error though.

**Comment by Jaex** [comment]
https://trac.ffmpeg.org/ticket/6375

FFmpeg errors never to do with ShareX.

**Comment by Scrxtchy** [comment]
Only thing that could be done is fixing any invalid inputs parameters that could be used. Which we had an attempt at fixing today which seems to have worked on a specific build of ffmpeg

**Comment by Betalord** [comment]
I want to confirm that Collisionsc's mpeg codec solved my problem. The thing doesn't crash anymore. Thanks!

**Comment by etssp** [comment]
Unfortunately, the link to http://collision.heroicstrike.org/ffmpeg.exe goes to a 58.6 MB file which is not an executable. It seems to be .7z file with some strange contents. Is it possible for someone to reupload Collisionc's file?

**Comment by Scrxtchy** [comment]
@etssp downloaded it myself, it is indeed a ffmpeg executable
Left, My local ffmpeg.exe, listing contents through 7z
Right, the one from Collisionc, also listing contents through 7z

![image](https://user-images.githubusercontent.com/12771982/38553058-6ba8339a-3d01-11e8-901d-4baf21763fef.png)

Just because 7z can open binaries, does not make them not binaries

**Comment by etssp** [comment]
Thanks Scrxtchy! I didn't know that executables could contain a list of files. I believe the problem is that we're speaking different languages. I'm using Windows XP 32-bit with ShareX 12.0, but this executable seems to be for a different platform, probably Linux. I get the following error message when I try to run the program "Access is denied." Is it possible to get a Windows XP 32-bit version? If not, where do I get the code to compile it and what parameters should I use? I'm getting the same error messages when I try to record long screen sessions. The version of ffmpeg.exe I'm currently using is ffmpeg-20160227-git-5156578-win32-static.

**Comment by rlabrecque** [comment]
I had this problem too, this solved it for me:

> Thanks to Scrxtchy's suggestion in IRC I fixed the audio and video being out of sync by "resetting" the sources, that is, putting them in None and then selecting the same ones again.

Using the default ffmpeg. Just needed to reset the sources.

**Comment by Jaex** [comment]
Resetting sources not makes any difference, it must be just coincidence.
But you can try installing latest recording devices from ShareX 12.1.1 which is supposed to fix audio sync issues.

**Comment by Venryx** [comment]
Just wanted to mention that in my case, the screen recorder actually worked fine -- the problem was in the conversion of the recorded MP4 to a GIF (since I was using the "Start/Stop screen recording (GIF)" action). I confirmed that the issue was in the conversion, by copying the MP4 file while it was being converted, and then viewing it separately.

Anyway, what I did to get around the issue is to change my hotkey to just record as an MP4, without converting it. And then if I want it in GIF format, I use this tool to easily convert it (it adds a convert-to-gif option in Windows Explorer's context-menu): https://github.com/Tichau/FileConverter

And then I upload it with the "Upload with ShareX" context-menu item in Windows Explorer (or ShareX's main window).
